packages/eslint-plugin/src/rules/return-await.ts

```diff
@@ -283,29 +288,53 @@ export default createRule({
         return;
       }
 
+      if (option === 'error-handling-correctness-only') {
+        if (!isAwait && affectsErrorHandling) {
+          context.report({
+            messageId: 'requiredPromiseAwait',
+            node,
+            // unconditional since always impacts error handling
```

Comment:
[Docs] Nitty nit: a bit more clear to read?

```suggestion
            // We always report here, as it always impacts error handling
```

Reply:
Made some further tweaks